packages/govern-console/src/containers/DAO/DaoHomePage.tsx

```diff
@@ -128,25 +128,29 @@ const DaoHomePage: React.FC = () => {
                   fetchMore={fetchMoreData}
                   actions={visibleActions}
                   isMore={IsMoreActions}
-                  identifier={daoName}
+                  daoName={daoName}
                 />
               )}
             />
             <ApmRoute
               exact
               path={`${path}finance`}
-              render={() => <div>Finance Component goes here...</div>}
+              render={() => (
+                <DaoFinancePage
+                  daoName={daoName}
+                  executorId={dao?.executor.id}
+                  token={dao?.token}
+                />
+              )}
             />
             <ApmRoute exact path={`${path}settings`} component={DaoSettings} />
-            <ApmRoute exact path={`${path}executions/:id`} component={ProposalDetails} />
-            <ApmRoute exact path={`${path}actions/new-execution`} component={NewExecution} />
-
-            {/* TODO: Operation not found page */}
-            <ApmRoute render={() => <div>Operation not found on dao. Go home?</div>} />
+            <ApmRoute exact path={`${path}actions/executions/:id`} component={ProposalDetails} />
+            <ApmRoute exact path={`${path}actions/new`} component={NewExecution} />
+            <ApmRoute render={() => history.push('/not-found')} />
           </Switch>
         </Suspense>
       </StyledGridItem>
-      {pathname === `${url}/settings` && (
+      {(pathname === `${url}/settings` || pathname === `${url}/actions/new`) && (
```

Comment:
Let's extract the path names to constants, so it is easier to do this type of comparison